### ISSUE TITLE
[BUG] 0 Graph score with empty tags

### ISSUE THREAD

**Issue by username116** [regorxxx/Search-by-Distance-SMP#27]
Hi, 

**Describe the bug**
Is it an expected result to have a 0 Graph score with values like this? : 

- Reference track:
Custom tag A: `Psychedelic Rock`
Custom tag B: `Funk`; `Lounge`; `Psychedelic Soul`

- Some tracks in the results with a 0 Graph score:
Custom tag A : (empty, no values)
Custom tag B : (empty, no values)

These custom tags A and B are the only ones with the "graph" type.

**Additional context**
I've lowered the setting `Set Search method` > `Minimum in case there are not enough tracks...` to `50`.

I've checked that these genres are present in the descriptors file.

I can provide a detailed description if required.

I've also seen 0 Graph scores with empty tags on both sides. But maybe that's what's expected.

Idea: show in the console next to the score, if the "graph" tags are empty for these tracks.

**Desktop**
 - OS: Windows 10
 - foobar2000 v2.1.1 x86, new portable installation
 - Spider Monkey Panel v1.6.1-mod, new panel without settings
 - Search-by-Distance-SMP downloaded today, commit https://github.com/regorxxx/Search-by-Distance-SMP/commit/954791f2b6e40cea66ce935cd40c229b88241fc5

**Comment by regorxxx** [comment]
That's a bug, the distance for not found genres (or in this case no genre) is Infinity.

At some point I deleted the check for no genres, should be fixed now (can not test it). 
- A reference track with no genre will throw a warning and report any other track as zero distance (uses WEIGHT as fallback and just compares scores). Why? Because it's a more logic expectation than having zero results using GRAPH.
- For reference track with genre, compared tracks with no genre will have an Infinity distance (and thus never used at output). Again the most logical expectation.

Let me know

EDIT: found other things in the process, wait some time.

**Comment by regorxxx** [comment]
Ok, should work now.

**Comment by username116** [comment]
Good to know it's a bug.
I tried the latest version. With the same reference track (the one that is Psychedelic Rock) the few tracks with empty "graph" tags have disappeared. The result is better: this time there are only identical genres (in those "graph" tags) in the playlist. So that's great.

With a reference track with empty "graph" tags, the result is the same: the 2 playlists are identical.
I've seen the warning. I understand the logic, but it can be a bit surprising to still have these "0 Graph" scores in the console, when it's a Weight method, and the "graph" tags can be empty on both sides. That's okay because I got your explanation. Maybe add in the warning that "all Graph scores will be 0"?

Thank you for this improvement and for the explanation.
